test/release.test.js

```diff
@@ -0,0 +1,360 @@
+        },
+      },
+    },
+  })
+
+  t.ok(
+    deleteReleaseStub.calledOnceWith({
+      owner: DEFAULT_ACTION_DATA.context.repo.owner,
+      repo: DEFAULT_ACTION_DATA.context.repo.repo,
+      release_id: 54503465,
+    })
+  )
+})
+
+tap.test('Should publish to npm without optic', async t => {
+  const { release, stubs } = setup()
+  await release({
+    ...DEFAULT_ACTION_DATA,
+    inputs: {
+      'npm-token': 'a-token',
+    },
+  })
+
+  t.ok(stubs.runSpawnStub.calledWith('npm', ['publish', '--tag', 'latest']))
+})
+
+tap.test('Should not publish to npm if there is no npm token', async t => {
+  const { release, stubs } = setup()
+  stubs.callApiStub.throws()
+
+  await release({
+    ...DEFAULT_ACTION_DATA,
+    inputs: {
+      'optic-token': 'optic-token',
+      'sync-semver-tags': 'true',
+    },
+  })
+
+  t.ok(
+    stubs.runSpawnStub.neverCalledWith('npm', ['publish', '--tag', 'latest'])
+  )
+  t.ok(
+    stubs.runSpawnStub.neverCalledWith('npm', [
+      'publish',
+      '--otp',
+      'otp123',
+      '--tag',
+      'latest',
+    ])
+  )
+})
+
+tap.test('Should publish to npm with optic', async t => {
+  const { release, stubs } = setup()
+  await release({
+    ...DEFAULT_ACTION_DATA,
+    inputs: {
+      'npm-token': 'a-token',
+      'optic-token': 'optic-token',
+    },
+  })
+
+  t.ok(
+    stubs.runSpawnStub.calledWith('npm', [
+      'publish',
+      '--otp',
+      'otp123',
+      '--tag',
+      'latest',
+    ])
+  )
+})
+
+tap.test('Should tag versions', async t => {
+  const { release, stubs } = setup()
+  await release({
+    ...DEFAULT_ACTION_DATA,
+    inputs: {
+      'npm-token': 'a-token',
+      'optic-token': 'optic-token',
+      'sync-semver-tags': 'true',
+    },
+  })
+
+  t.ok(stubs.tagVersionStub.calledWith('v5'))
+  t.ok(stubs.tagVersionStub.calledWith('v5.1'))
+})
+
+tap.test('Should call the release method', async t => {
+  const { release, stubs } = setup()
+  await release({
+    ...DEFAULT_ACTION_DATA,
+    inputs: {
+      'npm-token': 'a-token',
+      'optic-token': 'optic-token',
+      'sync-semver-tags': 'true',
+    },
+  })
+
+  t.ok(
+    stubs.callApiStub.calledWith(
+      {
+        endpoint: 'release',
+        method: 'PATCH',
+        body: {
+          version: 'v5.1.3',
+          releaseId: 54503465,
+        },
+      },
+      {
+        'npm-token': 'a-token',
+        'optic-token': 'optic-token',
+        'sync-semver-tags': 'true',
+      }
+    )
+  )
+})
+
+tap.test(
+  'Should not do anything if the user is not optic-release-automation[bot]',
+  async t => {
+    const { release, stubs } = setup()
+    await release({
+      ...DEFAULT_ACTION_DATA,
+      context: {
+        ...DEFAULT_ACTION_DATA.context,
+        payload: {
+          ...DEFAULT_ACTION_DATA.context.payload,
+          pull_request: {
+            ...DEFAULT_ACTION_DATA.context.payload.pull_request,
+            user: { login: 'not_the_correct_one' },
+          },
+        },
```

Comment:
Do you think this could be more clear for the reader?

```
const payload = clone(DEFAULT_ACTION_DATA)
payload.context.payload.pull_request.user = { login: 'not_the_correct_one' }
```

Reply:
You're absolutely right. I've updated all big nested objects to use a cloning func